### PR TITLE
authentication by user from db implemented; role model, dao and service added

### PR DESCRIPTION
**Pull request by DmGrt**

### CODE REVIEW

**Review by aleksandr-hrankin — APPROVED** [review]

File: src/main/java/com/spring/cinema/datainject/DataInjector.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.spring.cinema.service.RoleService;
+import com.spring.cinema.service.ShoppingCartService;
+import com.spring.cinema.service.UserService;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataInjector {
+    private static final String ADMIN = "ADMIN";
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/cinema/datainject/DataInjector.java]:
```suggestion
    private static final String ADMIN_ROLE_NAME = "ADMIN";
```

File: src/main/java/com/spring/cinema/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        User user;
+        UserBuilder builder;
+        if (optionalUser.isPresent()) {
+            user = optionalUser.get();
+            builder = org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(user.getPassword());
+            builder.roles(user.getRoles().stream().map(role -> role.getRoleName().toString())
+                    .toArray(String[]::new));
+        } else {
+            throw new UsernameNotFoundException("User not found!");
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/cinema/security/CustomUserDetailsService.java]: let's add email to exception message

File: src/main/java/com/spring/cinema/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        User user;
+        UserBuilder builder;
+        if (optionalUser.isPresent()) {
+            user = optionalUser.get();
+            builder = org.springframework.security.core.userdetails.User.withUsername(email);
+            builder.password(user.getPassword());
+            builder.roles(user.getRoles().stream().map(role -> role.getRoleName().toString())
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/cinema/security/CustomUserDetailsService.java]:
```suggestion
            builder.roles(user.getRoles().stream()
            .map(role -> role.getRoleName().toString()
            .toArray(String[]::new));
```

File: src/main/java/com/spring/cinema/security/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(email);
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/cinema/security/CustomUserDetailsService.java:22]: btw: 
to make code look better you can use orElseThrow here instead

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**